src/hooks/useDownload.ts

```diff
@@ -39,14 +39,16 @@ async function internal<T>({ src, dst, headers, ephemeral, logger }: DownloadOpt
     return () => memo ?? (memo = hash_key(src))
   })()
   const mtime_entry = () => hash().join("mtime")
+  const etag_entry = () => hash().join("etag")
 
   const { numpty } = useFlags()
   dst ??= hash().join(src.path().basename())
   if (src.protocol === "file:") throw new Error()
 
-  if (!ephemeral && mtime_entry().isFile() && dst.isReadableFile()) {
+  if (!ephemeral && mtime_entry().isFile() && dst.isReadableFile() && etag_entry().isFile()) {
```

Comment:
This means we'll only do these if BOTH etag and mtime are provided. ETAGs are much rarer than mtimes.

Reply:
Can we do this if any of mtime or etag condition met?

Reply:
yes, if any of them are met. I recommend reading up on how HTTP caching works, I do not have time to teach you. Thanks.